pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/RelToPlanNodeConverter.java

```diff
@@ -287,7 +294,12 @@ public static Set<String> getTableNamesFromRelRoot(RelNode relRoot) {
       // Calcite encloses table and schema names in square brackets to properly quote and delimit them in SQL
       // statements, particularly to handle cases when they contain special characters or reserved keywords.
       String tableName = qualifiedTableName.replaceAll("^\\[(.*)\\]$", "$1");
-      tableNames.add(tableName);
+      String[] split = tableName.split(", ");
+      if (split.length == 1) {
+        tableNames.add(tableName);
+      } else {
+        tableNames.add(DatabaseUtils.translateTableName(split[1], split[0]));
```

Comment:
I don't get this. Here database and table are separated by  `, `?

Reply:
`RelOptUtil.findAllTableQualifiedNames` returns the table name in the format `[foo, bar]` for a table `foo.bar` hence the split on `', '`